msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMSALController.java

```diff
@@ -136,31 +143,32 @@ public AcquireTokenResult acquireTokenSilent(MSALAcquireTokenSilentOperationPara
         }
 
         try {
-            //result = service.acquireTokenSilently(getSilentParameters(request));
-
-            // Path is broken by common/#303 - @kreedula & @shoatman to resolve
-            BrokerResult brokerResult = service.acquireTokenSilently(new BrokerRequest());
+            BrokerResult brokerResult = service.acquireTokenSilently(getBrokerRequest(parameters));
+            AcquireTokenResult acquireTokenResult = new AcquireTokenResult();
+            acquireTokenResult.setTokenResult(brokerResult);
+            if (brokerResult.isSuccessful() && brokerResult.getTokenResponse() != null) {
```

Comment:
Are we being defensive here, or is there some case where `true == brokerResult#isSuccessful()` _and_ the response is `null`?

Reply:
Yes. just being defensive, there should not be any case as such.